ipld/resolver/src/behaviour/membership.rs

```diff
@@ -188,14 +199,18 @@ impl Behaviour {
         let record = SignedProviderRecord::new(&self.local_key, self.subnet_ids.clone())?;
         let data = record.into_envelope().into_protobuf_encoding();
         let _msg_id = self.inner.publish(self.membership_topic.clone(), data)?;
+        self.last_publish_timestamp = Timestamp::now();
+        self.next_publish_timestamp = self.last_publish_timestamp + self.publish_interval.period();
+        self.publish_interval.reset(); // In case the change wasn't tiggered by the schedule.
         Ok(())
     }
 
     /// Mark a peer as routable in the cache.
     ///
     /// Call this method when the discovery service learns the address of a peer.
     pub fn set_routable(&mut self, peer_id: PeerId) {
-        self.provider_cache.set_routable(peer_id)
+        self.provider_cache.set_routable(peer_id);
+        self.publish_for_new_peer(peer_id);
```

Comment:
This assumes that when a new peer joins the topic all peers in their topic will share their membership, right? This will lead to bursts of messages with ever new joiner. I think this is fine with the time safe-guard you added, but we'll have to be careful as the size of the network increases as this may be too much and we may have to prune in some way who broadcasts their membership triggered by new joiners to spread the membership updates.

Reply:
You are completely right, it does lead to annoying bursts. Maybe they could choose some random time to publish, based on the cluster size. 

I just thought about an alternative, but it would need some rework: Instead of everyone publishing their own when they learn about a new guy, we could leave republishing only to the ones the new guy subscribes with. We don't have a point-to-point protocol here, but if we cached the incoming records as they are, we could publish them as they are, and I think the Gossipsub duplicate filter could make sure that everyone other than our new joiner would ignore the message. Does that sound plausible?

Reply:
> I think the Gossipsub duplicate filter could make sure that everyone other than our new joiner would ignore the message

Assuming this is in place (which I guess in Rust is also the case), I think this would work with a small caveat. Re-using messages may lead to them not being forwarded due to the filter, and not being propagated enough to reach the new joiner. It is probably best if we leave it is as-is for now and maybe think of an optimization if this becomes an issue, WDYT?